src/MapPanel.tsx

```diff
@@ -1,23 +1,26 @@
 import React, { Component } from 'react';
-import { PanelProps, createTheme, DataFrameView, getValueFormat } from '@grafana/data';
+import { PanelProps, createTheme, DataFrameView, getValueFormat, EventBus } from '@grafana/data';
 import { MapOptions } from './types';
 import { parseData } from './components/lib/dataParser';
 import { sanitizeTopology } from './components/lib/topologyTools';
 import './components/MapCanvas.component.js';
 import { PubSub } from './components/lib/pubsub.js';
+import { Unsubscribable } from 'rxjs';
 import { locationService } from '@grafana/runtime';
 import { resolvePath, setPath, LAYER_LIMIT } from "./components/lib/utils.js"
+
 export interface MapPanelProps extends PanelProps<MapOptions> {
   fieldConfig: any;
   options: MapOptions;
+  eventBus: EventBus;
 }
 
 export class MapPanel extends Component<MapPanelProps> {
   mapCanvas: any;
   lastOptions: any;
   theme: any;
   mapjsonCache: any;
-  subscriptionHandle: any;
+  subscriptionHandle?: Unsubscribable;
```

Comment:
Unnecessary type as `any`.

Reply:
This introduces a major dependency just for a type hint.

Reply:
Updated